apps/api/src/services/distributed-lock/index.ts

```diff
@@ -0,0 +1,135 @@
+import Redis from 'ioredis';
+import Redlock, { Lock, ResourceLockedError, Settings } from 'redlock';
+import { Logger } from '@nestjs/common';
+
+const LOG_CONTEXT = 'DistributedLock';
+
+const redisUrl = `${process.env.REDIS_HOST}:${process.env.REDIS_PORT}` || 'localhost:6379';
+
+interface ILockOptions {
+  resource: string;
+  ttl: number;
+}
+
+let distributedLock: Redlock;
+let lockCounter = 0;
+
+export const startup = (
+  settings: Settings = {
+    automaticExtensionThreshold: 500,
+    driftFactor: 0.01,
+    retryCount: 50,
+    retryDelay: 100,
+    retryJitter: 200,
+  }
+): void => {
+  if (distributedLock) {
+    return;
+  }
+
+  // TODO: Implement distributed nodes (at least 3 Redis instances)
+  const rawInstances = [redisUrl].filter((instance) => !!instance).map((url) => new Redis(url));
```

Comment:
This needs an architecture and resource change needed to discuss. We might be able to sustain the functionality right now for a while.
@Cliftonz @scopsy we need to discuss this.